From 3bba30fd23705dc8817b2eb0f28c9be03b8f7892 Mon Sep 17 00:00:00 2001
From: Alexander Couzens <lynxis@fe80.eu>
Date: Tue, 25 Jul 2017 15:39:36 +0200
Subject: common,atmodem: rename & move at_util_call_compare_by_id to common.c

at_util_call_compare_by_id is used by several modem drivers.
---
 drivers/atmodem/atutil.c        | 14 --------------
 drivers/atmodem/atutil.h        |  2 +-
 drivers/huaweimodem/voicecall.c |  6 +++---
 drivers/ifxmodem/voicecall.c    |  4 ++--
 drivers/stemodem/voicecall.c    |  2 +-
 src/common.c                    | 14 ++++++++++++++
 src/common.h                    |  1 +
 7 files changed, 22 insertions(+), 21 deletions(-)

diff --git a/drivers/atmodem/atutil.c b/drivers/atmodem/atutil.c
index fecae19e..9bf76ed0 100644
--- a/drivers/atmodem/atutil.c
+++ b/drivers/atmodem/atutil.c
@@ -77,20 +77,6 @@ gint at_util_call_compare_by_phone_number(gconstpointer a, gconstpointer b)
 				sizeof(struct ofono_phone_number));
 }
 
-gint at_util_call_compare_by_id(gconstpointer a, gconstpointer b)
-{
-	const struct ofono_call *call = a;
-	unsigned int id = GPOINTER_TO_UINT(b);
-
-	if (id < call->id)
-		return -1;
-
-	if (id > call->id)
-		return 1;
-
-	return 0;
-}
-
 GSList *at_util_parse_clcc(GAtResult *result, unsigned int *ret_mpty_ids)
 {
 	GAtResultIter iter;
diff --git a/drivers/atmodem/atutil.h b/drivers/atmodem/atutil.h
index 4afd21c8..862a0a14 100644
--- a/drivers/atmodem/atutil.h
+++ b/drivers/atmodem/atutil.h
@@ -53,7 +53,7 @@ typedef void (*at_util_sim_inserted_cb_t)(gboolean present, void *userdata);
 void decode_at_error(struct ofono_error *error, const char *final);
 gint ofono_call_compare_by_status(gconstpointer a, gconstpointer b);
 gint at_util_call_compare_by_phone_number(gconstpointer a, gconstpointer b);
-gint at_util_call_compare_by_id(gconstpointer a, gconstpointer b);
+gint ofono_call_compare_by_id(gconstpointer a, gconstpointer b);
 GSList *at_util_parse_clcc(GAtResult *result, unsigned int *mpty_ids);
 gboolean at_util_parse_reg(GAtResult *result, const char *prefix,
 				int *mode, int *status,
diff --git a/drivers/huaweimodem/voicecall.c b/drivers/huaweimodem/voicecall.c
index 9be8fb31..a4a48025 100644
--- a/drivers/huaweimodem/voicecall.c
+++ b/drivers/huaweimodem/voicecall.c
@@ -347,7 +347,7 @@ static void conf_notify(GAtResult *result, gpointer user_data)
 	ofono_info("Call setup: id %d", call_id);
 
 	l = g_slist_find_custom(vd->calls, GINT_TO_POINTER(call_id),
-				at_util_call_compare_by_id);
+				ofono_call_compare_by_id);
 	if (l == NULL) {
 		ofono_error("Received CONF for untracked call");
 		return;
@@ -384,7 +384,7 @@ static void conn_notify(GAtResult *result, gpointer user_data)
 	ofono_info("Call connect: id %d type %d", call_id, call_type);
 
 	l = g_slist_find_custom(vd->calls, GINT_TO_POINTER(call_id),
-				at_util_call_compare_by_id);
+				ofono_call_compare_by_id);
 	if (l == NULL) {
 		ofono_error("Received CONN for untracked call");
 		return;
@@ -428,7 +428,7 @@ static void cend_notify(GAtResult *result, gpointer user_data)
 				call_id, duration, end_status);
 
 	l = g_slist_find_custom(vd->calls, GINT_TO_POINTER(call_id),
-				at_util_call_compare_by_id);
+				ofono_call_compare_by_id);
 	if (l == NULL) {
 		ofono_error("Received CEND for untracked call");
 		return;
diff --git a/drivers/ifxmodem/voicecall.c b/drivers/ifxmodem/voicecall.c
index 60278d00..c02d27e8 100644
--- a/drivers/ifxmodem/voicecall.c
+++ b/drivers/ifxmodem/voicecall.c
@@ -135,7 +135,7 @@ static void xcallstat_notify(GAtResult *result, gpointer user_data)
 		return;
 
 	l = g_slist_find_custom(vd->calls, GINT_TO_POINTER(id),
-				at_util_call_compare_by_id);
+				ofono_call_compare_by_id);
 
 	if (l == NULL && status != CALL_STATUS_DIALING &&
 				status != CALL_STATUS_INCOMING &&
@@ -773,7 +773,7 @@ static void xcolp_notify(GAtResult *result, gpointer user_data)
 
 	l = g_slist_find_custom(vd->calls,
 				GINT_TO_POINTER(call_id),
-				at_util_call_compare_by_id);
+				ofono_call_compare_by_id);
 	if (l == NULL) {
 		ofono_error("XCOLP for unknown call");
 		return;
diff --git a/drivers/stemodem/voicecall.c b/drivers/stemodem/voicecall.c
index 19276111..849a9f87 100644
--- a/drivers/stemodem/voicecall.c
+++ b/drivers/stemodem/voicecall.c
@@ -462,7 +462,7 @@ static void ecav_notify(GAtResult *result, gpointer user_data)
 	 * If it doesn't exists we make a new one
 	 */
 	l = g_slist_find_custom(vd->calls, GUINT_TO_POINTER(id),
-				at_util_call_compare_by_id);
+				ofono_call_compare_by_id);
 
 	if (l)
 		existing_call = l->data;
diff --git a/src/common.c b/src/common.c
index b8df3785..f7cf9912 100644
--- a/src/common.c
+++ b/src/common.c
@@ -762,6 +762,20 @@ gint ofono_call_compare_by_status(gconstpointer a, gconstpointer b)
 	return 0;
 }
 
+gint ofono_call_compare_by_id(gconstpointer a, gconstpointer b)
+{
+	const struct ofono_call *call = a;
+	unsigned int id = GPOINTER_TO_UINT(b);
+
+	if (id < call->id)
+		return -1;
+
+	if (id > call->id)
+		return 1;
+
+	return 0;
+}
+
 const char *ofono_call_status_to_string(enum call_status status)
 {
 	switch (status) {
diff --git a/src/common.h b/src/common.h
index dc2c7005..9166ea34 100644
--- a/src/common.h
+++ b/src/common.h
@@ -187,4 +187,5 @@ gboolean is_valid_apn(const char *apn);
 
 gint ofono_call_compare(gconstpointer a, gconstpointer b);
 gint ofono_call_compare_by_status(gconstpointer a, gconstpointer b);
+gint ofono_call_compare_by_id(gconstpointer a, gconstpointer b);
 const char *ofono_call_status_to_string(enum call_status status);
-- 
cgit v1.2.1

